src/variety/squarejam.js

```diff
@@ -0,0 +1,168 @@
+//
+//  squarejam.js
+//
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["squarejam"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		inputModes: { edit: ["number", "clear"], play: ["border", "subline"] },
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					if (this.btn === "left" && this.isBorderMode()) {
+						this.inputborder();
+					} else {
+						this.inputQsubLine();
+					}
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart) {
+					this.inputqnum();
+				}
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true
+	},
+
+	Cell: {
+		minnum: 1,
+		maxnum: 999,
+
+		posthook: {
+			qnum: function() {
+				this.board.roommgr.setExtraData(this.room);
+			}
+		}
+	},
+
+	Board: {
+		hasborder: 1
+	},
+
+	AreaRoomGraph: {
+		enabled: true,
+
+		setExtraData: function(component) {
```

Comment:
Is the extra data used for anything? The `checkRoomSideLen` seems to do another scan of the numbers.

I think the `setExtraData` function can be removed, along with the call to this function in `Cell.posthook.qnum`.

Reply:
I got these from another rectangle-based room division puzzle, although I don't remember exactly which one. It was hard to tell what they were so I just left them in. I'll try removing and see if anything breaks